src/runner.nim

```diff
@@ -88,29 +109,55 @@ addOutputFormatter(formatter)
   var isBeforeFirstSuite = true
   var editedTestContents = "import streams, unittest_json\n"
 
-  for line in lines(conf.inputDir / testName):
+  for line in lines(paths.inputTest):
     if isBeforeFirstSuite and line.startsWith("suite"):
       isBeforeFirstSuite = false
       editedTestContents &= beforeTests
     editedTestContents &= line & '\n'
   editedTestContents &= afterTests
-  writeFile(result, editedTestContents)
-  copyFile(getAppDir().parentDir() / "src" / "unittest_json.nim",
-           tmpDir / "unittest_json.nim")
-  createDir(conf.outputDir)
+  writeFile(paths.tmpTest, editedTestContents)
 
-proc run*(testPath: string): int =
+proc prepareFiles*(paths: Paths) =
+  ## Copies the solution file and an edited version of the input test file.
+  createDir(paths.outResults.parentDir())
+  copyFile(paths.inputSol, paths.tmpSol)
+  copyEditedTest(paths)
+  copyFile(paths.unittestJson, paths.tmpUnittestJson)
+
+proc run*(paths: Paths): int =
   ## Compiles and runs the file in `testPath`. Returns its exit code.
   # Use `startProcess` here, not `execCmd`.
   result = -1
 
-  let args = @["c", "-r", "--styleCheck:hint", "--skipUserCfg:on",
-               "--verbosity:0", "--hint[Processing]:off", testPath]
+  let args = @["c", "--styleCheck:hint", "--skipUserCfg:on", "--verbosity:0",
+               "--hint[Processing]:off", paths.tmpTest]
 
   var
     p = startProcess("nim", args = args, options = {poStdErrToStdOut, poUsePath})
     outp = outputStream(p)
     line = newStringOfCap(120).TaintedString
+    compilationMessages: string
+
+  while true:
+    if outp.readLine(line):
+      stdout.writeLine(line)
+      if compilationMessages.len != 0:
+        compilationMessages &= "\\n"
+      compilationMessages &= line
+    else:
+      result = peekExitCode(p)
+      if result != -1:
+        break
+  close(p)
```

Comment:
```suggestion
let
  (output, exitCode) = execCmdEx("nim c --styleCheck:hint --skipUserCfg:on --verbosity:0 --hint[Processing]:off " & paths.tempTest)
...
```

Reply:
I think this is more succinct and easier to handle in my opinion

Reply:
Done. I did it the previous way because it seemed like there could be cases where it's nice to print output as it happens.